rails-new.md

```diff
@@ -1,77 +1,129 @@
+
+- *gemfile* : This file lists the gem dependencies for your application. A gem is a Ruby software package, and a Gemfile allows you to manage your project's software needs. 
+- *app*: The app directory is where your main application code lives. This includes the models, controllers, views, assets, helpers, and mailers that make up the application itself. Rails gives you some application-level boilerplate for the MCV model to start out in files like `app/models/application_record.rb`, `app/controllers/application_controller.rb`, and `app/views/layouts/application.html.erb`.
+- *config*: This directory contains your application's configuration settings.
+- *config/routes*: Your application's route declarations live in this file.
+- *config/application.rb*: General settings for your application components are located in this file. 
+- *config/environments*: This directory is where configuration settings for your environments live. Rails includes three environments by default: `development`, `test`, and `production`. 
+- *config/database*: Database configuration settings live in this file, which is broken into four sections: `default`, `development`, `production`, and `test`. Thanks to the Gemfile that came with the `rails new bnb-library --database=postgresql`, which included the `pg` gem, our `config/database.yml` file has its adapter parameter set to postgresql already, specifying that we will use an postgresql database with this application. 
+- *db*: This folder includes a directory for database migrations called migrate, along with the `schema.rb` and `seeds.rb` files. `schema.db` contains information about your database, while `seeds.rb` is where you can place seed data for the database.
+
+## Step 3 - Starting Your Application
+
+We are going to want to interact with our app in the browser, but first, we must setup our database.
+
+In our file titled `.docker-development-vars.rb` we added a *DATABASE_URL* which is going to tell our rails application a few things. Here is the breakdown...
+```
+postgres://DATABASE_USERNAME:DATABASE_PASSWORD@DATABASE_HOST:PORT/DATABASE_NAME
+```
+This won't be any help to us if we don't tell our application to use this variable, so let's head over to `database.yml` and under `development` we will add the *DATABASE_URL* environment variable.
+
+```
+development:
+  <<: *default
+  database: bnb_library_development
+  url: <%= ENV['DATABASE_URL'] %>
+```
+Now setup and migrate the database
+```
+dip rake db:setup
+dip rake db:migrate
+```
+You can expect an output similar to 
+```
+Starting bnb-library_postgres_1 ... done
+Running via Spring preloader in process 19
+
+Created database 'bnb_library_development'
+/app/db/schema.rb doesn't exist yet. Run `bin/rails db:migrate` to create it, then try again.
+```
+This is because we have not created any tables or corresponding migrations... yet! For now, let's start your application's server and see the landing page rails has created for us.
```

Comment:
This bit feels confusing. Why run `db:migrate` if we haven't created migrations? But I think rails generates ActiveStorage migrations and maybe one other system migration already, so that should run.

Not sure what to do about it.

Reply:
In testing this, it looks like no migrations are added in initializing a new app, so I will remove the migration step